Resolves #59015, Scheduler: Add support for EBS types t3, r5, & z1d

cuppett opened the pull request:
The new t3, r5, r5d and z1 need matched as well according to this:

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/volume_limits.html

T3, C5, C5d, M5, M5d, R5, R5d, and z1d instances support a maximum of 28 attachments, and every instance has at least one network interface attachment. If you have no additional network interface attachments on these instances, you could attach 27 EBS volumes.

fixes #59015

jessfraz committed:
pkg/kubelet/apis/cri/runtime: add masked_paths and readonly_paths

generate runtime protobufs

Signed-off-by: Jess Frazelle <acidburn@microsoft.com>

jessfraz committed:
ProcMount: add api options and feature gate

Signed-off-by: Jess Frazelle <acidburn@microsoft.com>

jessfraz committed:
ProcMount: update staging types

Signed-off-by: Jess Frazelle <acidburn@microsoft.com>

mikedanese committed:
svcacct: pass pod information in user.Info.Extra() when available

Fixes https://github.com/kubernetes/kubernetes/issues/59670

szuecs committed:
fix #51135 make CFS quota period configurable, adds a cli flag and config option to kubelet to be able to set cpu.cfs_period and defaults to 100ms as before.
It requires to enable feature gate CustomCPUCFSQuotaPeriod.

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

juanvallejo committed:
add prototype sorting for table rows

wgliang committed:
Add CSI volume attributes for kubectl describe pv

jessfraz committed:
update jsonlog path for updated vendor

Signed-off-by: Jess Frazelle <acidburn@microsoft.com>

jessfraz committed:
ProcMount: add dockershim support

Signed-off-by: Jess Frazelle <acidburn@microsoft.com>

Commit:
Add docker images for node perf testing workloads.

Commit:
fix usage string for the kubectl logs command

Even though the use of an inline [CONTAINER] name is still accepted,
this does not match what the documentation or man page says. This commit
aligns the usage string that is displayed when the kubectl logs command
is called with more than one container name (with the use of flag or
not).

riverzhang committed:
Fix some descriptions for kubeadm

from `v1alpha1` to `v1alpha2`.

sttts committed:
kube-controller-manager: add test server and test serving

Commit:
Add validation for kube-scheduler

adding validation for componentconfig

adding validation to cmd kube-scheduler

Add support for ipv6 in IsValidSocketAddr function

updating copyright date in componentconfig/validation/validation.go

updating copyright date in componentconfig/validation/validation_test.go

adding validation for cli options

adding BUILD files

updating validate function to return []errors in cmd/kube-scheduler

ok, really returning []error this time

adding comments for exported componentconfig Validation functions

silly me, not checking structs along the way :'(

refactor to avoid else statement

moving policy nil check up one function

rejigging some deprecated cmd validations

stumbling my way around validation slowly but surely

updating according to review from @bsalamat

- not validating leader election config unless leader election is enabled
- leader election time values cannot be zero
- removing validation for KubeConfigFile
- removing validation for scheduler policy

leader elect options should be non-negative

adding test cases for renewDeadline and leaseDuration being zero

fixing logic in componentconfig validation :sweat_smile:

removing KubeConfigFile reference from tests as it was removed in master

https://github.com/kubernetes/kubernetes/commit/2ff9bd6699ccdfb1f46c4aeda01518990ff71eda

removing bogus space after var assignment

adding more tests for componentconfig based on feedback

making updates to validation because types were moved on master

update bazel build

adding validation for staging/apimachinery

adding validation for staging/apiserver

adding fieldPaths for staging validations

moving staging validations out of componentconfig

updating test case scenario for staging/apimachinery

./hack/update-bazel.sh

moving kube-scheduler validations from componentconfig

./hack/update-bazel.sh

removing non-negative check for QPS

resourceLock required

adding HardPodAffinitySymmetricWeight 0-100 range to cmd flag help section

Commit:
Update resource-consumer usage

The e2e tests are using version 1.3 of resource-consumer from
gcr.io/kubernetes-e2e-test-images since [1]. In addition, HPA
works fine for a deployment with --requrests flag.
So this updates the README.md for fitting current condition.

[1]: https://github.com/kubernetes/kubernetes/blob/master/test/utils/image/manifest.go#L86

sttts committed:
kube-controller-manager: enable delegated authz/authn if secure port is enabled

sttts committed:
cloud-controller-manager: enable delegated authz/authn if secure port is enabled

sttts committed:
kube-controller-manager: enable secure ports 10257, deprecate insecure port

sttts committed:
cloud-controller-manager: enable secure ports 10258, deprecate insecure port

sttts committed:
kube-controller-manager: enable secure loopback

grayluck committed:
Add namespace for (cluster)role(binding) cloud-provider.

Change the addonmanager mode to be from reconcile to EnsureExists.

Commit:
support continueToken for inconsistent list

Commit:
- Added multi-arch support.
- Used multi-stage builds.

mkumatag committed:
Sync peer-finder code from contrib repo

mikedanese committed:
reload token file for InClusterConfig every 5 minutes

neolit123 committed:
pkg/util/net: use a more descriptive error in getAllDefaultRoutes()

Change the error output of getAllDefaultRoutes() so that it includes
information on which files were probed for the IP routing tables
even if such files are obvious.

Introduce a new error type which can be used to figure out of this
error is exactly of the "no routes" type.

neolit123 committed:
kubeadm: update auto-generated BUILD files

neolit123 committed:
kubeadm: fix the air-gapped and offline support issues

1) Do not fail in case a bind address cannot be obtained

If netutil.ChooseBindAddress() fails looking up IP route tables
it will fail with an error in which case the kubeadm config
code will hard stop.

This scenario is possible if the Linux user intentionally disables
the WiFi from the distribution settings. In such a case the distro
could empty files such files as /proc/net/route and ChooseBindAddress()
will return an error.

For improved offline support, don't error on such scenarios but instead
show a warning. This is done by using the NoRoutesError type.
Also default the address to 0.0.0.0.

While doing that, prevent some commands like `init`, `join` and also
phases like `controlplane` and `certs` from using such an invalid
address.

Add unit tests for the new function for address verification.

2) Fallback to local client version

If there is no internet, label versions fail and this breaks
air-gapped setups unless the users pass an explicit version.

To work around that:
- Remain using 'release/stable-x.xx' as the default version.
- On timeout or any error different from status 404 return error
- On status 404 fallback to using the version of the client via
kubeadmVersion()

Add unit tests for kubeadmVersion().

Co-authored-by: Alexander Kanevskiy <alexander.kanevskiy@intel.com>

stewart-yu committed:
[kube-controller-manager] auto-generated file about run godeps

sjenning committed:
cloudprovider: aws: return true on existence check for stopped instances

chanjarster committed:
fix https://github.com/kubernetes/kubernetes/issues/67561

jessfraz committed:
vendor: update docker cadvisor winterm

This vendor change was purely for the changes in docker to allow for
setting the Masked and Read-only paths.

See: moby/moby#36644

But because of the docker dep update it also needed cadvisor to be
updated and winterm due to changes in pkg/tlsconfig in docker

See: google/cadvisor#1967

Signed-off-by: Jess Frazelle <acidburn@microsoft.com>

chanjarster committed:
fix https://github.com/kubernetes/kubernetes/issues/67561 bump version

ahmad-diaa committed:
use topologyPairsMaps for inter pod affinity/anti-affinity maps

jsafrane committed:
Introduce new `CSIDriver` CRD API Object

jsafrane committed:
Add new csi-api repo to staging

jsafrane committed:
Introduce new `CSINodeInfo` CRD API Object

jsafrane committed:
Pass new CSI API Client and informer to Volume Plugins

Commit:
generated

Random-Liu committed:
Support extra prune resources in kube-addon-manager.

Signed-off-by: Lantao Liu <lantaol@google.com>

mgdevstack committed:
transitioning container-runtime from e2e_node to e2e/common

mbohlool committed:
update generated files

saad-ali committed:
Generated code for new APIs

tnozicka committed:
Generalize kubectl rollout StatusViewer interface

tnozicka committed:
Switch kubectl rollout status to UntilWithSync to avoid premature
timeouts

Katharine committed:
Build relevant binaries with coverage.

Katharine committed:
Add runtime coverage support.

mbohlool committed:
Refactor addmission webhook hook client to a util package

saad-ali committed:
Automatically install CRDs during controller init

liztio committed:
Two implmentations of cert renewal

Katharine committed:
Pass KUBE_BUILD_WITH_COVERAGE through to docker instance.

tizhou86 committed:
Add function comment to fix golint error in cmd/kube-scheduler/app.

sttts committed:
cloud-controller-manager: enable secure loopback

sttts committed:
cloud-controller-manager: add test server

sttts committed:
controller-managers: generalize authn/z test to cloud-controller-manager

krzysztof-jastrzebski committed:
Change CPU sample sanitization in HPA.
Ignore samples if:
- Pod is beeing initalized - 5 minutes from start defined by flag
    - pod is unready
    - pod is ready but full window of metric hasn't been colected since
    transition
- Pod is initialized - 5 minutes from start defined by flag:
    - Pod has never been ready after initial readiness period.

rajansandeep committed:
prep for 1.12

gnufied committed:
Add gnufied as approver for attach/detach controller

Hopefully has reviewed and made enough fixes in this
area to understand the code thoroughly.

smarterclayton committed:
Reduce the minwidth of the cli table printer

For a few columns we want to have a smaller width and 10 is excessive.

ahmad-diaa committed:
addressed reviewer comments

liztio committed:
Build artifacts

liztio committed:
Mechanism for renewing a certificate based on an existing certificate

Katharine committed:
Add autogenerated BUILD files.

stewart-yu committed:
add OWNERS file

andyzhangx committed:
add mixed protocol support for azure load balancer

add logging

fix lb issue

fix comments

make code more clean

krzysztof-jastrzebski committed:
Increase Horizontal Pod Autoscaler update frequency (every 30s -> every
15s).

wojtek-t committed:
Allow for configuring etcd servers

figo committed:
add test to verify vsphere cloud provider report node hostname

as in pull #67922 has modify vsphere cloud provider to report
node hostname, this patch is to add the test for it.

also fix an issue at InstanceID(), it suppose to return
cloudprovider.InstanceNotFound when vm not found, after the fix,
test TestInstance() can pass

damemi committed:
Fix conversion for autoscaling/v1 ObjectMetricSource and add MetricIdentifier fuzzer

Selectors in ObjectMetricSource's weren't being persisted through roundtrip conversions, and this wasn't caught because we had no fuzzer testing MetricIdentifier selectors

neolit123 committed:
cluser/addons: add labels to fluentd owner files

liztio committed:
Actually renew certificates (using on-disk CAs)

Commit:
Put fluentd back to host network

wojtek-t committed:
Fix unnecessary too-old-errors from watch cache

krzysztof-jastrzebski committed:
Update auto generated files.

Commit:
Add --server-dry-run flag to `kubectl apply`

liztio committed:
Add `renew all` command

riverzhang committed:
Fix how to use ipvs mode by kubeadm

jessfraz committed:
address comments

Signed-off-by: Jess Frazelle <acidburn@microsoft.com>

jessfraz committed:
make update

Signed-off-by: Jess Frazelle <acidburn@microsoft.com>

sttts committed:
kube-controller-manager: disable authn/z on insecure port

This is the old behaviour and we did not intent to change it due to enabled authn/z in general.
As the kube-apiserver this sets the "system:unsecured" user info.

sttts committed:
cloud-controller-manager: disable authn/z on insecure port

This is the old behaviour and we did not intent to change it due to enabled authn/z in general.
As the kube-apiserver this sets the "system:unsecured" user info.

luxas committed:
Automated package reference rename

BenTheElder committed:
add make targets for building server images

Katharine committed:
Improve bash formatting.

Katharine committed:
Be sure we delete the dummy tests.

vishh committed:
Adding GCE node termination handler as an optional addon.
This step is a pre-requisite for auto-deploying that addon in GKE.

Signed-off-by: Vishnu kannan <vishnuk@google.com>

tallclair committed:
Cleanup PodSecurityPolicy AllowPrivEsc tests

k82cn committed:
Taint node in paralle.

Signed-off-by: Klaus Ma <klaus1982.cn@gmail.com>

krzysztof-jastrzebski committed:
Replace scale down forbidden window

    Replacement is scale down stabilization window. HPA will scale down only
    to max of recommendations it made during that window. More details in

    https://docs.google.com/document/d/1IdG3sqgCEaRV3urPLA29IDudCufD89RYCohfBPNeWIM

andyzhangx committed:
support cross resource group for azure file

fix test failure

bertinatto committed:
Promote mount propagation to GA

Commit:
Move image locality to default priority functions

Commit:
Fix cloning image states from node info

Commit:
Add integration tests for image locality priority

krzysztof-jastrzebski committed:
Update auto generated files.

serathius committed:
Increase metrics-server scrape frequency to 30s

dixudx committed:
as hostpathtype owner, adds myself to OWNERS file

wojtek-t committed:
Fix retrying in ipam controller

k82cn committed:
Wait for Scheduler cache empty.

Signed-off-by: Klaus Ma <klaus1982.cn@gmail.com>

shyamjvs committed:
Use random backoff for retries in cloud-cidr-allocator

cuppett committed:
Resolves #59015, extends existing regex to cover t3, r5(d) & z1d instance types

From current AWS documentation:

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/volume_limits.html

T3, C5, C5d, M5, M5d, R5, R5d, and z1d instances support a maximum of
28 attachments, and every instance has at least one network interface
attachment. If you have no additional network interface attachments on
these instances, you could attach 27 EBS volumes.

cuppett committed:
Resolves #59015, extends existing regex to cover t3, r5(d) & z1d instance types

From current AWS documentation:

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/volume_limits.html

T3, C5, C5d, M5, M5d, R5, R5d, and z1d instances support a maximum of
28 attachments, and every instance has at least one network interface
attachment. If you have no additional network interface attachments on
these instances, you could attach 27 EBS volumes.

cuppett commented:
/sig aws
/sig storage

cuppett commented:
/assign gnufied

gnufied commented:
/ok-to-test

cuppett commented:
/retest

shyamjvs committed:
GC kubemark image after cluster starts up

tallclair committed:
Hide & warn on GA & deprecated feature gates

Katharine committed:
Address review comments.

misterikkit commented on pkg/scheduler/algorithm/predicates/predicates.go:
```diff
@@ -378,7 +378,7 @@ func getMaxVolumeFunc(filterName string) func(node *v1.Node) int {
 
 func getMaxEBSVolume(nodeInstanceType string) int {
-	if ok, _ := regexp.MatchString("^[cm]5.*", nodeInstanceType); ok {
+	if ok, _ := regexp.MatchString("^[cmr]5.*|t3|z1d", nodeInstanceType); ok {
```
Having the same regex in two places that need to be kept in sync makes me uneasy. Also, for someone who is not in the know, how would I learn whether this regex is correct? Are there test cases for the supported AWS volume types?

cuppett replied:
There is an existing test from the previous PR #66397 for m5/c5. Updated commit description and title. Thanks!

misterikkit reviewed (commented):
I'll leave it to sigs aws and storage to approve of the regex change.

I would prefer to see a git commit message as descriptive as your PR description, and I would also like for the title/top line to be more descriptive. e.g. instead of "resolves #xyz" it could be something like, "scheduler: add support for EBS types t3, r5, & z1d"

spiffxp committed:
Add spiffxp to OWNERS files in bash-heavy dirs

I'm comfortable approving changes in hack/, I think I still need
to build up a corpus of reviews in build/ and cluster/ before I'm
comfortable asking for those rights

BenTheElder committed:
unify server images bash build

mikedanese committed:
cluster: delete some kube-push remanents

Katharine committed:
Add KUBE_COVERAGE_FLUSH_INTERVAL to set flush interval.

mikedanese committed:
make deps-approvers the approvers of sample-cli-plugin/Godeps

Katharine committed:
Update stale comment.

Katharine committed:
Improve error behaviour of package coverage.

Katharine committed:
Address shell-related comments.

dims committed:
Update echoserver version used to 2.2

Change-Id: Ic1dcb2c64ac682ca601ab2589fd6af70d4e09620

Katharine committed:
Add owners.

Katharine committed:
Change owners.

tallclair committed:
Add RuntimeClass read permission for nodes

cblecker committed:
Add Godeps OWNERS for csi-api

luxas committed:
Refactor the kube-controller-manager ComponentConfig structs to they can be moved out

luxas committed:
Fixup cmd/*controller-manager code after struct changes. Co-authored by @stewart-yu

luxas committed:
autogenerated

luxas committed:
Move the kubelet's external types to k8s.io/kubelet

luxas committed:
Add skeleton new repo files

luxas committed:
autogenerated go code, godeps, bazel and gofmt

esevan committed:
Fixup openstack cloud provider loadbalancer deletion error

This change enables ```getLoadBalancer``` to return the loadbalancer even if no floating ip is associated to the VIP port of the loadbalancer.

Signed-off-by: Eunsoo Park <esevan.park@gmail.com>

tnozicka committed:
Update Bazel

stewart-yu committed:
[kube-controller-manager] fix violation and golint failure

stewart-yu committed:
[kube-controller-manager] fix some reference from cmd/*-controller-manager about kubeControllerManagerConfiguration

stewart-yu committed:
[kube-controller-manager] auto-generated file

feiskyer committed:
Fix panic when processing http response

stewart-yu committed:
[kube-controller-manager] fix some reference from pkg/apis/componentconfig about kube-controller-manager api

stewart-yu committed:
[kube-controller-manager] just only remove struct and default about KubeControllerManagerConfiguration from pkg/apis/componentconfig

stewart-yu committed:
[kube-controller-manager] create package to hold kube-controller-manager component api
